src/shadowbox/server/shared_metrics.spec.ts

```diff
@@ -121,6 +121,35 @@ describe('OutlineSharedMetricsPublisher', () => {
 
       publisher.stopSharing();
     });
+
+    it('reports ASN metrics correctly', async () => {
+      const clock = new ManualClock();
+      const serverConfig = new InMemoryConfig<ServerConfigJson>({serverId: 'server-id'});
+      const usageMetrics = new ManualUsageMetrics();
+      const metricsCollector = new FakeMetricsCollector();
+      const publisher = new OutlineSharedMetricsPublisher(
+        clock,
+        serverConfig,
+        null,
+        usageMetrics,
+        metricsCollector
+      );
+      publisher.startSharing();
+
+      usageMetrics.locationUsage = [
+        {country: 'DD', asn: 999, inboundBytes: 44},
```

Comment:
Can we have a test where the same country has multiple ASNs?
Also, I believe technically the same ASN can be in multiple countries. It may be worth testing that as well.